2020-07-12 16:09:30,903 ERROR: Exception on /user/shuichiro [GET] [in /home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/cromz/learning/microblog/app/routes.py", line 76, in user
    return render_template('user.html', user=user, posts=posts, form=form)
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/home/cromz/learning/microblog/venv/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/cromz/learning/microblog/app/templates/user.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/cromz/learning/microblog/app/templates/base.html", line 30, in top-level template code
    {% block content%}{% endblock %}
  File "/home/cromz/learning/microblog/app/templates/user.html", line 22, in block "content"
    {% elif not current_user.if_following(user) %}
jinja2.exceptions.UndefinedError: 'app.models.User object' has no attribute 'if_following'
2020-07-12 16:10:06,773 INFO: Microblog startup [in /home/cromz/learning/microblog/app/__init__.py:43]
2020-07-12 16:14:49,062 INFO: Microblog startup [in /home/cromz/learning/microblog/app/__init__.py:43]
